### DISCUSSION TITLE
Filament has a critical problem. Can't integrate external api in Login and other pages.

### DISCUSSION THREAD

**Discussion by awesome1128** [filamentphp/filament#15104] [General]
I am at first use with filament, but I found the critical problem.
I have all my data on baserow.io, so I want to use it as my database, not laravel database such as mysql etc.
First of all, I tried to build Login page, but there is no solution to integrate baserow api with Login page.
So, I think, this is super critical problem that filament has.
Please let me know if someone know something about this.
Thank you.

**Comment by leandrocfe** [comment]
This isn't an issue with Filament itself, as Filament is built on the TALL stack. To customize the [login/auth process](https://filamentphp.com/docs/3.x/panels/users#authentication-features), you'll need to implement the changes using [Livewire](https://livewire.laravel.com/docs/quickstart).

I recommend reviewing the Livewire and Laravel documentation to better understand how they work.

**Reply by awesome1128** [reply]
Hello @leandrocfe 
Thank you for your quick reply.
I have reviewed [login/auth process](https://filamentphp.com/docs/3.x/panels/users#authentication-features) for a while, so yes it makes sense.

But another question.
For example, I can see this function in Filament document.
```
public function panel(Panel $panel): Panel
{
    return $panel
        // ...
        ->loginRouteSlug('login')
        ->registrationRouteSlug('register')
        ->passwordResetRoutePrefix('password-reset')
        ->passwordResetRequestRouteSlug('request')
        ->passwordResetRouteSlug('reset')
        ->emailVerificationRoutePrefix('email-verification')
        ->emailVerificationPromptRouteSlug('prompt')
        ->emailVerificationRouteSlug('verify');
}
```

As you can see, there are many functions called with $panel.
For example, how can I where is loginRouteSlug function etc?
Is there any document that explain functions of Filament?
Thank you.

**Reply by leandrocfe** [reply]
check [php api doc](https://filamentphp.com/api/3.x/index.html)

**Reply by awesome1128** [reply]
Awesome!
Thank you so much!
